### ISSUE TITLE
InteractionReceiver

### ISSUE THREAD

**Issue by OptiHub** [Microsoft/MixedRealityToolkit-Unity#2808]
**Does this affect the legacy HoloToolkit (master) or the Mixed Reality Toolkit (mrtk_release)?**
HoloToolkit

**Describe the bug**
I updated all my Interactions in my Project with the InteractionReceiver. Problem is now when I intantiate an object from a script (Object is not in the receiver), I cannot shoot any interaction on this object. 

**To Reproduce**
1. Make an InteractionReceiver like explained [here](https://github.com/Microsoft/MixedRealityToolkit-Unity/blob/master/Assets/HoloToolkit-Examples/UX/Readme/README_InteractableObjectExample.md#receiver)
2. Also make a Dialog like explained [here](https://github.com/Microsoft/MixedRealityToolkit-Unity/blob/master/Assets/HoloToolkit-Examples/UX/Readme/README_DialogsExample.md) with an OK Button for closing the Dialog. Try to instantiate the Dialog in the script.

**Expected behavior**
When I click on OK the Dialog should close

**Actual behavior**
Nothing happens

**Unity Editor Version**
Unity2018.2.5f1

**Mixed Reality Toolkit Release Version**
Downloaded the latest Master (08-27-2018)

**Additional context**
I also tested this with other objects. All my interactable Objects are now not working when the receiver is added to the project (unless it is added to the receiver, but I think this is not working with an Instantiation)

**Comment by keveleigh** [comment]
@radicalad Is this anything you have insight into after your port of `InteractionReciever`?

**Comment by radicalad** [comment]
This makes sense. The InteractionReceiver’s list is either created in editor or during run time. If you can get the reference to the button, you can add it to the receiver manually by interactables.Add() (the list is public). To me, this sounds like a issue with dialogue (you should be able to access the button instances). I can take a look tomorrow AM.

**Comment by radicalad** [comment]
Just to clarify, are you saying you have an InteractionReceiver instantiating a Dialog object, and the popped dialog is not closing? I can't repro your issue.


`public class DialogueTest : InteractionReceiver
{

    //fill dialogPrefab in the inspector:
    
    public Dialog dialogPrefab;
    
    // Use this for initialization
    void Start ()
    {
        //then call:
        Dialog dialog = Dialog.Open(dialogPrefab.gameObject, DialogButtonType.OK, "Testing", "Test message");
    }
	
	// Update is called once per frame
	void Update () {
		
	}

    protected override void InputDown(GameObject obj, InputEventData eventData)
    {
        Dialog dialog = Dialog.Open(dialogPrefab.gameObject, DialogButtonType.OK, "Testing", "Test message");
    }
`
this works as expected, clicking an object the `InteractionReceiver` is linked to throws a Dialog, that dialog closes when clicked.

**Comment by OptiHub** [comment]
Thank you @radicalad for testing.

I copied your code in my scene and it was not working, so I created a new scene and there it worked. 

I found out that the mistake was in the toolbar I used from the Holotoolkit. I think I changed something in the settings there, because when I deleted the toolbar and added a new one everything was working just fine. I don't know what it was exactly but for now the issue is gone. 

Thank you for helping me!
